sdk/src/protocols/uniswap_v3/index.ts

```diff
@@ -0,0 +1,112 @@
+import { queryTokens, findToken } from "./utils"
+import { EthToken } from "./types"
+import { allowErc20Approve, oneOf } from "../../conditions"
+import { contracts } from "../../../eth-sdk/config"
+import { allow } from "zodiac-roles-sdk/kit"
+import { c, Permission } from "zodiac-roles-sdk"
+import ethInfo from "./_ethInfo"
+import { BigNumber } from "ethers"
+import { NotFoundError } from "../../errors"
+
+const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000"
+
+export const eth = {
+  deposit: async ({
+    targets,
+    tokens,
+    avatar,
+  }: {
+    /** Position NFT token IDs to allow depositing into. If unspecified, all positions owned by avatar can be managed that are in any pair of the specified `tokens`.
+     *
+     * **Attention:** If the avatar has approved the Uniswap NFT Position contract to spend tokens other than the ones specified in `tokens`, the role will be able to increase and decrease any existing positions' liquidity in these tokens as well.
+     */
+    targets?: string[]
+    /** Positions can be minted for any pair of the specified `tokens`. If unspecified, minting of new positions won't be allowed. */
+    tokens?: (EthToken["address"] | EthToken["symbol"])[]
+    avatar: `0x${string}`
+  }) => {
+    if (!targets && !tokens) {
+      throw new Error("Either `targets` or `tokens` must be specified.")
+    }
+
+    const nftIds =
+      targets &&
+      targets.map((target) => {
+        try {
+          return BigNumber.from(target)
+        } catch (e) {
+          // could not be parsed as BigNumber
+          throw new NotFoundError(`Invalid NFT ID: ${target}`)
+        }
+      })
+
+    const tokensForTargets = nftIds && (await queryTokens(nftIds))
+
+    const mintTokenAddresses =
+      tokens?.map((addressOrSymbol) => findToken(ethInfo, addressOrSymbol)) ||
+      []
+
+    const permissions: Permission[] = [
+      ...allowErc20Approve(tokensForTargets || [], [
+        contracts.mainnet.uniswap_v3.positions_nft,
+      ]),
+      ...allowErc20Approve(mintTokenAddresses, [
+        contracts.mainnet.uniswap_v3.positions_nft,
+      ]),
+      {
+        ...allow.mainnet.uniswap_v3.positions_nft.mint(
+          {
+            recipient: avatar,
```

Comment:
```suggestion
            recipient: c.avatar,
```

This makes it fully dynamic (avatar address is looked up when checking permissions onchain) and allows us to entirely get rid of this function's `avatar` param.

Reply:
You're totally right!